### PR TITLE
[FIX] manifest-version-format: Use real dot from regex

### PR DESCRIPTION
**Pull request by moylop260**
Fix https://github.com/OCA/pylint-odoo/issues/152

cc @naglis

### CODE REVIEW

**Comment by naglis** [comment]
LGTM from technical point of view, however, I find that now it becomes a bit ambiguous on how to define the valid versions in the config file - I think I will find myself writing `8.0` instead of `8\.0`.

Also, I think it breaks the existing version checks inside _pylint_odoo_ checkers, eg.:

https://github.com/OCA/pylint-odoo/blob/42418ab69664e06c0b834016d667e88ab970f99e/pylint_odoo/checkers/modules_odoo.py#L800-L801

And [the ones](https://github.com/versada/pylint-odoo/blob/a8b57fc4c1b04a89ad77c61c2d95ff347446f203/pylint_odoo/checkers/modules_odoo.py#L854-L857) from #146.

**Comment by moylop260** [comment]
@naglis you have a good point.
Good catch!
Thank you

Could you review the new changes, please?

**Comment by naglis** [comment]
@moylop260 nice, did not know about `re.escape` :+1:

However, now the issue with string interpolation in `ConfigParser` which I have described in #152 remains. For example, if I have the following `.pylintrc` file and use it with _pylint-odoo_:
```ini
[ODOOLINT]
manifest_version_format=(%(valid_odoo_versions)s)\.\d+\.\d+\.\d+$
valid_odoo_versions=10.0
```
Then due to the string interpolation in `ConfigParser` by the time [`formatversion()`](https://github.com/OCA/pylint-odoo/blob/a14ddca6f0d57e603c55f60b0128b129ad5fb81e/pylint_odoo/checkers/no_modules.py#L657) is called `self.config.manifest_version_format` will already be formatted to `(10.0)\.\d+\.\d+\.\d+$` and the `re.escape`'d version numbers will simply be ignored by the `%` function (non-existent placeholders do not raise an exception). So version numbers like `10+0.1.0.0` will still be able to slip by with such a config file :/

**Comment by moylop260** [comment]
I got it.
Thank you

I can reproduce it using:

`pylint --rcfile=pylint_manifest_version.cfg --load-plugins=pylint_odoo pylint_odoo/test_repo/broken_module`

```conf
[MESSAGES CONTROL]
disable=all
enable=manifest-version-format

[ODOOLINT]
manifest_version_format=(%(valid_odoo_versions)s)\.\d+\.\d+\.\d+
valid_odoo_versions=10.0
```

 - screen shot 2017-09-05 at 10 20 36

The `manifest_version_format` is using magical interpolation previously like as your comment: https://github.com/OCA/pylint-odoo/issues/152

Let me see if there is a method to pre-process the parameter from definition

**Comment by moylop260** [comment]
You won man
We need bypass the magic interpolation then using other parsing string like as the "hated" `format` is fine for this case 😃 

Thank you @naglis for your patient to understand the issue

**Comment by naglis** [comment]
@moylop260 no problem, it's a tricky one to explain. Thank you for your work on this issue 🙇

**Comment by moylop260** [comment]
Could you give me your approval, please?

**Comment by naglis** [comment]
I am not a reviewer, but you have my approval 👍

**Comment by moylop260** [comment]
Could you review @lasley @pedrobaeza @dreispt @JesusZapata please?

File: pylint_odoo/checkers/no_modules.py [hunk 1]
```diff
@@ -655,9 +655,10 @@ def camelize(self, string):
         return re.sub(r"(?:^|_)(.)", lambda m: m.group(1).upper(), string)
 
     def formatversion(self, string):
-        self.config.manifest_version_format_parsed = \
-            self.config.manifest_version_format % dict(
-                valid_odoo_versions='|'.join(self.config.valid_odoo_versions))
+        self.config.manifest_version_format_parsed = (
+            self.config.manifest_version_format.format(
+                valid_odoo_versions='|'.join(
+                    map(re.escape, self.config.valid_odoo_versions))))
```
Comment (lasley) [inline pylint_odoo/checkers/no_modules.py]: Can we use some intermediary variables here? I'm having a bit of trouble following all the nested method calls IMO

**Comment by moylop260** [comment]
@lasley 
What about now?

**Pull request merged**